androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java

```diff
@@ -266,255 +274,215 @@ public void test_02_create_public_link_folder_all_enabled()
         SystemClock.sleep(WAIT_CONNECTION_MS);
 
         //Check the status of the sharing options
-        onView(withId(R.id.shareViaLinkEditPermissionSwitch)).check(matches(isChecked()));
-        if (isSupportedFileListing()) {
-            onView(withId(R.id.shareViaShowFileListingSwitch)).check(matches(isEnabled()));
-            onView(withId(R.id.shareViaShowFileListingSwitch)).check(matches(isChecked()));
-        }
-        onView(withId(R.id.shareViaLinkPasswordSwitch)).check(matches(isChecked()));
-        onView(withId(R.id.shareViaLinkExpirationSwitch)).check(matches(isChecked()));
+        onView(withId(R.id.shareViaLinkEditPermissionReadOnly)).check(matches(isChecked()));
+
+        onView(withId(R.id.shareViaLinkPasswordSwitch)).check(matches(not(isChecked())));
+        onView(withId(R.id.shareViaLinkExpirationSwitch)).check(matches(not(isChecked())));
 
-        onView(withId(R.id.cancelAddPublicLinkButton)).perform(scrollTo(), click());
+        onView(withId(R.id.cancelButton)).perform(scrollTo(), click());
         SystemClock.sleep(WAIT_CONNECTION_MS);
 
         //Remove the link
         deleteLink();
 
-        Log_OC.i(LOG_TAG, "Test Share Public All Enabled Passed");
+        Log_OC.i(LOG_TAG, "Test Share Public with Download/View");
 
     }
 
     /**
-     *  TEST CASE: Share public a folder and gets the link
-     *  PASSED IF: Link correctly copied in clipboard
+     *  TEST CASE: Share publicly a folder woth DownloadV/View/Upload permission
+     *  PASSED IF: Link created and visible in share view with Download/View/Upload option
+     *
      */
     @Test
-    public void test_03_get_link()
+    public void test_03_create_public_link_download_view_permission()
```

Comment:
What about renaming the test to `test_03_create_public_link_download_view_upload_permission`?